auth/kerberos/src/krb.cpp

```diff
@@ -724,7 +724,7 @@ std::pair<int, std::string> get_gmsa_krb_ticket( std::string domain_name,
  */
 std::string get_ticket_expiration( std::string klist_ticket_info )
 {
-    std::regex pattern(".+\n.{21}(.{19}).+");
+    std::regex pattern(".+\n[^\s]+\s+.{8}\s+([^\s]+\s+.{8}).+");
```

Comment:
Could you add a comments about the regex

Reply:
```
.+\n         	 Skip all characters up to the first newline.
[^\s]+     	 Next, skip one or more of non whitespace characters
\s+.{8}\s+       Next, whitespace(s) and a length of 8 characters followed by whitespace(s)
([^\s]+\s+.{8})  Next, match for group 1, one or more of non whitespace followed by whitespace(s) and 8 characters
.+               Next, ignore all remaining characters after group 1
```

Reply:
A test like a klist output would be good, we can add that here.